File: freemenot/freemenot.py
import dsl

def getContainer():
   c = dsl.MyContainer()
   o1 = dsl.MyObject("This is a label")
   o1.thisown = 0
   c.addObject(o1)
   return c

c = getContainer()
print (c.getNrOfObjects())

#if the thisown flag for objects in the getContainer function
#is equal to 1, the following call return an undefined object
#If the flag is equal to 0, the following call will return a valid object
a = c.getObject(0)
print (a.getLabel())

File: freemenot/freemenot.cpp
#include "freemenot.h"
#include <iostream>
using namespace std;

namespace mine
{

/* MyObject source */
MyObject::MyObject(const string& lbl)
:
label(lbl)
{
    cout<<"In object ctor"<<endl;
}

MyObject::~MyObject()
{
    cout<<"In object dtor"<<endl;
}

string MyObject::getLabel()
{
    return label;
}

/* MyContainer source */
MyContainer::MyContainer() { cout<<"In container ctor"<<endl; }

MyContainer::~MyContainer()
{
    cout<<"In container dtor"<<endl;
    for(unsigned int i = 0; i < mObjects.size(); i++)
    {
        delete mObjects[i];
    }
}

int MyContainer::getNrOfObjects()
{
    return mObjects.size();
}

void MyContainer::addObject(MyObject* o)
{
    mObjects.push_back(o);
}

MyObject* MyContainer::getObject(unsigned int i)
{
    return mObjects[i];
}

}
